msal/application.py

```diff
@@ -809,6 +820,7 @@ def acquire_token_by_username_password(
             - A successful response would contain "access_token" key,
             - an error response would contain "error" and usually "error_description".
         """
+        self.authority.initialize()
```

Comment:
Nit: Suggest to move this line to line 829.5 i.e. right where it is needed. So that if/when in future we would refactor the authority logic for ADFS, we would be less likely to forget this otherwise-separated `self.authority.initialize()`.